lib/apiservers/engine/backends/image.go

```diff
@@ -344,12 +345,39 @@ func (i *Image) PullImage(ctx context.Context, image, tag string, metaHeaders ma
 	}
 	//*****
 
+	// create url from hostname
+	hostnameURL, err := url.Parse(ref.Hostname())
+	if err != nil || hostnameURL.Hostname() == "" {
+		hostnameURL, err = url.Parse("//" + ref.Hostname())
+		if err != nil {
+			log.Infof("Error parsing hostname %s during registry access: %s", ref.Hostname(), err.Error())
+		}
+	}
+
 	options := imagec.Options{
 		Destination: os.TempDir(),
 		Reference:   ref.String(),
 		Timeout:     imagec.DefaultHTTPTimeout,
 		Outstream:   outStream,
-		RegistryCAs: RegistryCertPool,
+	}
+
+	// Check if url is contained within set of whitelisted or insecure registries
+	vchConfig := VchConfig()
+	insecureOk := RegistrySetContains(vchConfig.InsecureRegistries, hostnameURL)
+	whitelistOk := RegistrySetContains(vchConfig.RegistryWhitelist, hostnameURL)
+	if len(vchConfig.RegistryWhitelist) > 0 && !whitelistOk {
+		err = fmt.Errorf("Access denied to unauthorized registry (%s) while VCH is in whitelist mode", hostnameURL.Hostname())
+		log.Errorf(err.Error())
+		sf := streamformatter.NewJSONStreamFormatter()
+		outStream.Write(sf.FormatError(err))
+		return nil
+	}
+	options.InsecureAllowHTTP = insecureOk
+
+	if insecureOk {
```

Comment:
We don't remove the system root CA pool.  Nil means the user probably didn't pass in a cert for this server (although it's a possibility) so we check the system certs.  If that doesn't work, we bypass TLS.